packages/@aws-cdk/aws-pipes-alpha/lib/source.ts

```diff
@@ -100,3 +103,48 @@ export interface ISource {
    */
   grantRead(grantee: IRole): void;
 }
+
+/**
+ * Sources that support a dead-letter target.
+ */
+export abstract class SourceWithDlq implements ISource {
```

Comment:
My only comment in this file, besides a repeat on `instanceof` is the lack of consistency between using `Dlq` and `DeadLetter` in the contract. I'd prefer one or the other be chose. Usually I think it's better to write it out.

Reply:
I changed everything to `...DeadLetterTarget`.